devicetypes/ezex/somfy-blind-controller-ezex.src/somfy-blind-controller-ezex.groovy

```diff
@@ -0,0 +1,123 @@
+            }
+        }
+        standardTile("contPause", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "pause", label:"", icon:'st.sonos.pause-btn', action:'pause', backgroundColor:"#cccccc"
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 1) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        valueTile("ShadeLevel", "device.level", width: 4, height: 1) {
+            state "level", label: 'Shade is ${currentValue}% up', defaultState: true
+        }
+        controlTile("levelSliderControl", "device.level", "slider", width:2, height: 1, inactiveLabel: false) {
+            state "level", action:"switch level.setLevel"
+        }
+
+        main "windowShade"
+        details(["windowShade", "contPause", "ShadeLevel", "levelSliderControl","refresh"])
+    }
+}
+
+private getCLUSTER_WINDOW_COVERING() { 0x0102 }
+private getATTRIBUTE_POSITION_LIFT() { 0x0008 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    def parseMap = zigbee.parseDescriptionAsMap(description)
+    log.info "LOG-INFO-Shade: In parse(), description:- ${description}"
+    def attrs = parseMap.additionalAttrs
+    log.info "LOG-INFO-Shade: In parse(), additionalAttrs: - ${attrs}"
+    if ( attrs != null ) {
+        attrs.each { attr ->
+            if(parseMap.clusterInt == CLUSTER_WINDOW_COVERING && attr.attrInt == ATTRIBUTE_POSITION_LIFT) {
+                if (attr.value == "64") { //open
+                    sendEvent(name: "windowShade", value: "open")
+                    sendEvent(name: "level", value: "100")
+                } else if (attr.value == "00") { //closed
+                    sendEvent(name: "windowShade", value: "closed")
+                    sendEvent(name: "level", value: "0")
+                } else {
+                    log.info "Partially open state event"
+                    sendEvent(name: "windowShade", value: "partially open")
+                    sendEvent(name: "level", value: zigbee.convertHexToInt(attr.value))
+                }
+            }
+        }
+    }
+}
+
+def close() {
+    log.info "LOG-INFO-Shade: close()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x01)
+}
+
+def open() {
+    log.info "LOG-INFO-Shade: open()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x00)
+}
+
+def setLevel(data) {
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x05, zigbee.convertToHexString(data, 2))
+}
+
+def pause() {
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x02)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return refresh()
+}
+
+def refresh() {
+    def cmds =  zigbee.readAttribute(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT)
```

Comment:
nit: extra space after `=`

Reply:
I noticed this had been marked "resolved" but does not appear to be resolved? (There is an extra space after the '='.)

Reply:
fixed.